Sources/NIOCore/CurrentEventLoopPromise.swift

```diff
@@ -0,0 +1,44 @@
+public struct CurrentEventLoopPromise<Value> {
+    public let wrapped: EventLoopPromise<Value>
+    
+    @inlinable public var futureResult: CurrentEventLoopFuture<Value> {
+        wrapped.futureResult.iKnowIAmOnTheEventLoopOfThisFuture()
+    }
+    
+    @inlinable
+    init(_ promise: EventLoopPromise<Value>) {
+        self.wrapped = promise
+    }
+}
+
+#if swift(>=5.6)
+@available(*, unavailable)
+extension CurrentEventLoopPromise: Sendable {}
+#endif
+
+extension EventLoopPromise {
+    @inlinable public func iKnowIAmOnTheEventLoopOfThisPromise(
```

Comment:
Same nitpick as below, it is trying a bit _too_ hard to be funny and does not explain why I'd be doing this.

Can this instead be `assumeOnPromiseEventLoop()` or `assumeOnEventLoopOfPromise()` or if you need to be super explicit `assume...OrCrash()`? The "iKnow" sound cute but don't help understand and parse the code while reading it.